internal/drain-config/drainConfig.go

```diff
@@ -45,17 +45,16 @@ func NewDrainConfigManager(conf *config.Config, log logger.Logger) (*drainConfig
 		log.Errorw("db setup", "error", err)
 		return nil, fmt.Errorf("db setup: %v", err)
 	}
-	if err := migrate(db); err != nil {
+	if err = migrate(db); err != nil {
 		log.Errorw("db migrations", "error", err)
-		return nil, fmt.Errorf("db migrations: %v", err)
 	}
 	return &drainConfigManager{
 		log:  log,
 		conf: conf,
 		db:   db,
 
 		done: &atomic.Bool{},
-	}, nil
```

Comment:
Can we add a doc on the producer function explaining this special scenario (returning a non-nil manager with a non nil error)?

Reply:
done

Reply:
Have we consider returning an error but having the gateway app ignoring it?

Reply:
yes, we are returning error and ignoring it in gateway